camayoc/tests/qcs/cli/test_sources.py

```diff
@@ -0,0 +1,944 @@
+        '            "name": "{}"\r\n'
+        '        }}\r\n'
+        '    \],\r\n'
+        '    "hosts": \[\r\n'
+        '        "{}"\r\n'
+        '    \],\r\n'
+        '    "id": \\d+,\r\n'
+        '    "name": "{}",\r\n'
+        '    "port": {},\r\n'
+        '    "source_type": "{}"\r\n'
+        '}}\r\n'
+        .format(cred_name, hosts, name, port, source_type)
+    )
+
+    with open('hosts_file', 'w') as handler:
+        handler.write(new_hosts + '\n')
+
+    qpc_source_edit = pexpect.spawn(
+        'qpc source edit --name {} --hosts {}'
+        .format(name, 'hosts_file')
+    )
+    qpc_source_edit.logfile = BytesIO()
+    assert qpc_source_edit.expect(
+        'Source "{}" was updated'.format(name)) == 0
+    assert qpc_source_edit.expect(pexpect.EOF) == 0
+    qpc_source_edit.close()
+    assert qpc_source_edit.exitstatus == 0
+
+    if new_hosts.endswith('0/24'):
+        new_hosts = new_hosts.replace('0/24', '\[0:255\]')
+    elif new_hosts.endswith('[1:100]'):
+        new_hosts = new_hosts.replace('[1:100]', '\[1:100\]')
+    source_show(
+        {'name': name},
+        '{{\r\n'
+        '    "credentials": \[\r\n'
+        '        {{\r\n'
+        '            "id": \\d+,\r\n'
+        '            "name": "{}"\r\n'
+        '        }}\r\n'
+        '    \],\r\n'
+        '    "hosts": \[\r\n'
+        '        "{}"\r\n'
+        '    \],\r\n'
+        '    "id": \\d+,\r\n'
+        '    "name": "{}",\r\n'
+        '    "port": {},\r\n'
+        '    "source_type": "{}"\r\n'
+        '}}\r\n'
+        .format(cred_name, new_hosts, name, port, source_type)
+    )
+
+
+def test_edit_hosts_negative(
+        isolated_filesystem, qpc_server_config, source_type):
+    """Edit the hosts of a not created source entry.
+
+    :id: 802d4a66-b0ee-4351-bff4-52469ed023f4
+    :description: Edit the hosts of a not created source entry.
+    :steps: Run ``qpc source edit --name <invalidname> --hosts <newhosts>``
+    :expectedresults: The command should fail with a proper message.
+    """
+    cred_name = utils.uuid4()
+    name = utils.uuid4()
+    hosts = '127.0.0.1'
+    new_hosts = '127.0.0.{}'.format(random.randint(2, 255))
+    invalid_name = utils.uuid4()
+    cred_add(
+        {
+            'name': cred_name,
+            'username': utils.uuid4(),
+            'password': None,
+            'type': source_type,
+        },
+        [(CONNECTION_PASSWORD_INPUT, utils.uuid4())],
+    )
+
+    qpc_source_add = pexpect.spawn(
+        'qpc source add --name {} --cred {} --hosts {} --type {}'
+        .format(name, cred_name, hosts, source_type)
+    )
+    assert qpc_source_add.expect('Source "{}" was added'.format(name)) == 0
+    assert qpc_source_add.expect(pexpect.EOF) == 0
+    qpc_source_add.close()
+    assert qpc_source_add.exitstatus == 0
+
+    qpc_source_edit = pexpect.spawn(
+        'qpc source edit --name {} --hosts {}'
+        .format(invalid_name, new_hosts)
+    )
```

Comment:
If the test is just checking that I can't edit a source that does not exist, I'm not sure if you really need to bother creating a valid source in the first part of this test. Are you just wanting there to be at least on source on the server at the time that you try and edit your fake source? I don't think it hurts anything but may just be extra code we don't need.

Reply:
This is the negative test I talked about on https://github.com/quipucords/camayoc/pull/116#discussion_r162344632. This test  needs to be update so it can check if hosts can't be edited with invalid values and for that a valid source will be required.

After updating this I will check any other test that may require update in terms of being a good test.